usl_pipeline/study_area_uploader/tests/transformers/test_study_area_transformers.py

```diff
@@ -98,3 +123,185 @@ def test_transform_shape_file_with_polygon_masks():
         assert soil_classes[0][1] == 8
         assert soil_classes[1][0].equals(bbox_polygon(1, 0, 3, 2))
         assert soil_classes[1][1] == 9
+
+
+def test_prepare_and_upload_study_area_files_no_boundaries_buildings_only():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        work_dir = pathlib.Path(temp_dir)
+        elevation_input_file_path = work_dir / "input_elevation.tif"
+        prepare_elevation_data(
+            elevation_input_file_path,
+            [
+                [0.0, 1.0, 2.0, 3.0, 4.0],
+                [5.0, 6.0, 7.0, 8.0, 9.0],
+                [10.0, 11.0, 12.0, 13.0, 14.0],
+                [15.0, 16.0, 17.0, 18.0, 19.0],
+                [20.0, 21.0, 22.0, 23.0, 24.0],
+            ],
+        )
+        buildings_input_file_path = work_dir / "buildings.shp"
+        prepare_shape_file(
+            buildings_input_file_path,
+            [(bbox_polygon(2, 1, 4, 3), 1)],
+        )
+        study_area_bucket = mock.MagicMock()
```

Comment:
When working with mocks, it's best to spec them
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock
This prevents you from calling methods on the mock object that the actual class it is mocking doesn't have. So here, it'd be
```
from google.cloud import storage

study_area_bucket = mock.MagicMock(spec=storage.Bucket)
boundaries_blob = mock.MagicMock(spec=storage.Blob)
```
etc.

Reply:
Done

Reply:
Also I noticed that your code doesn't completely follow these suggestions:
https://github.com/UrbanSystemsLab/climateiq-cnn/commit/349941c5d65a04dcc6b1d91bbab91890d956f933#diff-b91f014c9f83a783e5e5dea25e4d37c17686a840d34690c2be676b4e8340b136R44-R50

Reply:
Yes, I should be spec-ing those.